server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -230,4 +228,71 @@ protected void checkDuplicates(
     updatedDuplicateOptions.putAll(customDuplicateOptions);
     checkDuplicateEntries(ctx, foundEntries, updatedDuplicateOptions);
   }
+
+  /**
+   * Flags errors for rule lists passed as parameters if there are multiple instances of mutually
+   * exclusive options.
+   *
+   * @param options Options checked to insert into error message
+   * @param rules Lists of TerminalNode to iterate through
+   * @return Number of TerminalNode instances found
+   */
+  protected int checkHasMutuallyExclusiveOptions(String options, List<TerminalNode>... rules) {
+    String tokenType = null;
+    int occurances = 0;
+    for (List<TerminalNode> option : rules) {
+      if (option.isEmpty()) continue;
+      for (TerminalNode rule : option) {
+        if (rule == null) continue;
+        if (tokenType == null) {
+          tokenType = option.get(0).getText().toUpperCase();
```

Comment:
For terminals, the following should work as well, without stringifying the token itself
`rule.getSymbol().getType()`.